p2p/protocol/adv/adv_test.go

```diff
@@ -0,0 +1,54 @@
+package adv_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/filecoin-project/go-legs/p2p/protocol/adv"
+	"github.com/filecoin-project/go-legs/test"
+	"github.com/ipfs/go-datastore"
+	dssync "github.com/ipfs/go-datastore/sync"
+	_ "github.com/ipld/go-ipld-prime/codec/dagjson"
+	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
+	"github.com/ipld/go-ipld-prime/node/basicnode"
+	blhost "github.com/libp2p/go-libp2p-blankhost"
+	swarmt "github.com/libp2p/go-libp2p-swarm/testing"
+)
+
+func TestFetchLatestAdv(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	publisher := blhost.NewBlankHost(swarmt.GenSwarm(t, ctx))
+	client := blhost.NewBlankHost(swarmt.GenSwarm(t, ctx))
```

Comment:
Can you use `libp2p.New` and avoid the additional swarm/blankhost dependencies like https://github.com/filecoin-project/go-legs/blob/main/legs_test.go#L29 ?

Reply:
sure! Does this dependency show up for users of the library? (even if it's just used in the test?)

Reply:
go.mod is getting better at this, but it's still not great and test dependencies do often still make it in to the dependency tree.